workflow/controller/dag_test.go

```diff
@@ -614,6 +617,7 @@ func TestDagAssessPhaseContinueOnExpandedTaskVariables(t *testing.T) {
 	woc := newWorkflowOperationCtx(wf, controller)
 
 	woc.operate()
+	woc = newWorkflowOperationCtx(woc.wf, controller)
```

Comment:
why do we need to create a new context here? we can just use woc.operate() which will have updated wf.